### PR TITLE
Using WorkManager for Offline Task Scheduling

### LINKED ISSUE #967: Refactor from Android Job to WorkManager
### Chore

**Description**

Work Manager as introduced in Google IO 18 offers additional features than Evernote Android Job (which is currently being used in the project). 

1. Being able to chain work 
2. The ability to be opportunistic on work with constraints.

This is aimed at replacing Evernote-Android-Job by Work Manager.

**Would you like to work on the issue?**

  Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #967

Changes: Using Work Manager to handle offline task scheduling instead of Evernote-Android-Job.

Screenshots for the change:

### CODE REVIEW

**Comment by Masquerade0097** [comment]
@iamareebjamal Please review.

File: app/build.gradle [hunk 1]
```diff
@@ -204,4 +203,8 @@ dependencies {
     testImplementation "org.robolectric:robolectric:${versions.roboelectric}"
     testImplementation "org.robolectric:shadows-multidex:${versions.roboelectric}"
     testImplementation "org.robolectric:shadows-supportv4:${versions.roboelectric}"
+
+    //WorkManager
+
+    implementation "android.arch.work:work-runtime:${versions.work_manager}"
```
Comment (iamareebjamal) [inline app/build.gradle]: Move it below livedata dependency

File: app/proguard-rules.pro [hunk 2]
```diff
@@ -61,14 +61,5 @@
...
 
--dontwarn com.evernote.android.job.gcm.**
--dontwarn com.evernote.android.job.util.GcmAvailableHelper
-
--keep public class com.evernote.android.job.v21.PlatformJobService
--keep public class com.evernote.android.job.v14.PlatformAlarmService
--keep public class com.evernote.android.job.v14.PlatformAlarmReceiver
--keep public class com.evernote.android.job.JobBootReceiver
--keep public class com.evernote.android.job.JobRescheduleService
-
```
Comment (iamareebjamal) [inline app/proguard-rules.pro]: Check if there's proguard requirement in WorkManager

File: app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeCheckInWork.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Inject
+    AttendeeRepositoryImpl attendeeRepository;
+
+    @NonNull
+    @Override
+    public WorkerResult doWork() {
+        DaggerAppComponent.create().inject(this);
+
+        Timber.d("Running work");
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeCheckInWork.java]: Tell which work

File: app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeCheckInWork.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        Constraints constraints = new Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build();
+
+        OneTimeWorkRequest request = new OneTimeWorkRequest.Builder(AttendeeCheckInWork.class)
+            .setConstraints(constraints)
+            .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, WorkRequest.MIN_BACKOFF_MILLIS, TimeUnit.MILLISECONDS)
+            .build();
+
+        WorkManager.getInstance().beginUniqueWork("checkin baby", ExistingWorkPolicy.REPLACE, request).enqueue();
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeCheckInWork.java]: Replace the tag and move to constant

File: app/proguard-rules.pro (hunk 2 above)
Comment (Masquerade0097) [reply app/proguard-rules.pro]: I couldn't find any progaurd requirement for WorkManager.

**Comment by Masquerade0097** [comment]
@iamareebjamal Done !!

**Pull request merged**